pkg/apis/resource/basic_view.go

```diff
@@ -205,6 +202,14 @@ func (r *PatchRequest) Validate() error {
 		return err
 	}
 
+	// Set computedAttributes.
+	patched.EnvironmentID = r.Environment.Refer.ID()
```

Comment:
1. `r.Environment.Refer` cannot guarantee that it always receives an ID from the request, I think you can use `r.Environemnt.ID` or query with `patched` directly.
2. I saw we used `patched.EnvironmentID` previously, why do we need to change it here?